src/components/GenericTable/ColumnVisibilityButton.test.tsx

```diff
@@ -0,0 +1,197 @@
+import { render, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { axe } from "jest-axe";
+import ColumnVisibilityButton from "./ColumnVisibilityButton";
+
+const columns: Column[] = [
+  { field: "name", label: "Name", hideable: false },
+  { field: "age", label: "Age" },
+  { field: "email", label: "Email" },
+];
+
+const getColumnKey = (column: Column) => column.fieldKey ?? column.field;
+const getColumnLabel = (column: Column) => column.label;
+
+let columnVisibilityModel: { [key: string]: boolean };
+let setColumnVisibilityModel: jest.Mock;
+
+type Column = {
+  field: string;
+  fieldKey?: string;
+  label: string;
+  hideable?: boolean;
+};
+
+describe("Accessibility", () => {
+  beforeEach(() => {
+    columnVisibilityModel = {
+      name: true,
+      age: true,
+      email: true,
+    };
+    setColumnVisibilityModel = jest.fn((model) => {
+      columnVisibilityModel = model;
+    });
+  });
+
+  it("should not have accessibility violations when closed", async () => {
+    const { container } = render(
+      <ColumnVisibilityButton
+        columns={columns}
+        columnVisibilityModel={columnVisibilityModel}
+        getColumnKey={getColumnKey}
+        getColumnLabel={getColumnLabel}
+        onColumnVisibilityModelChange={setColumnVisibilityModel}
+      />
+    );
+
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("should not have accessibility violations when open", async () => {
+    const { container, getByTestId } = render(
+      <ColumnVisibilityButton
+        columns={columns}
+        columnVisibilityModel={columnVisibilityModel}
+        getColumnKey={getColumnKey}
+        getColumnLabel={getColumnLabel}
+        onColumnVisibilityModelChange={setColumnVisibilityModel}
+      />
+    );
+
+    userEvent.click(getByTestId("column-visibility-button"), null, { skipHover: true });
```

Comment:
Just annotating this for future reference, mostly because I'll forget...

I made the changes to line 62-66. This test was throwing an error in CI because the MUI tooltip was getting triggered by the userEvent.click simulation, which also includes a hover event (as a user would need to hover over a button to click it). However, when a test finishes before all of the events wrap up, Jest throws an act warning/error (super annoying but necessary).

In this case, I just disabled the hover event. There's other ways to address this though.

Reply:
Thanks for fixing this for me, there are a lot of jest errors that are just new to me and can be difficult to debug. I was unaware jest complains about this sort of thing. I appreciate the fix and the explanation!